### PR TITLE
add device detail header visual component

### PR DESCRIPTION
**Pull request by drcoolsanjeev**
### Type of change

- Enhancement / new feature

### Description
add visual component for device detail header
image

### CODE REVIEW

File: console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.test.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React from "react";
+import { render } from "@testing-library/react";
+import { MemoryRouter } from "react-router";
+import { DeviceDetailHeader } from "./DeviceDetailHeader";
+
+describe("<DeviceDetailHeader />", () => {
+  it("should renders a iot project header", () => {
```
Comment (rkpattnaik780) [inline console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.test.tsx:12]: should renders -> should render

File: console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  Split,
+  SplitItem,
+  Title,
+  Flex,
+  FlexItem,
+  DropdownItem,
+  DropdownPosition,
+  PageSection,
+  PageSectionVariants,
+  Badge,
```
Comment (rkpattnaik780) [inline console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:12]: This variable is not being used.

File: console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+}
+
+export const project_header_styles = StyleSheet.create({
+  flex_left_border: {
+    paddingLeft: "1em",
+    borderLeft: "0.05em solid",
+    borderRightColor: "lightgrey"
+  },
+  address_space_icon_margin: {
+    backgroundColor: "#EC7A08",
```
Comment (rkpattnaik780) [inline console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:38]: address_space_icon_margin, address_icon_align are not being used and badge_style are not being used.

File: console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
+import React, { useState } from "react";
```
Comment (rkpattnaik780) [inline console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:1]: License is missing.

File: console/console-init/ui/src/stories/DeviceDetail.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React from "react";
+import { MemoryRouter } from "react-router";
+import { DeviceFilter } from "modules/device";
```
Comment (rkpattnaik780) [inline console/console-init/ui/src/stories/DeviceDetail.stories.tsx:8]: Unused variable.

File: console/console-init/ui/src/stories/DeviceList.stories.tsx [hunk 6]
```diff
@@ -20,7 +20,7 @@ import { IRowData } from "@patternfly/react-table";
 import { DeviceDetailNavigation } from "modules/device-detail";
 
 export default {
-  title: "Device"
+  title: "Device List Table"
```
Comment (rkpattnaik780) [inline console/console-init/ui/src/stories/DeviceList.stories.tsx:23]: The stories in this file are not limited to `Device List Table`. It includes everything in the device page.

File: console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+  onEdit: () => void;
+  onDelete: () => void;
+  onClone: () => void;
+}
+
+export const project_header_styles = StyleSheet.create({
+  flex_left_border: {
+    paddingLeft: "1em",
+    borderLeft: "0.05em solid",
+    borderRightColor: "lightgrey"
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:35]: Can you please check, if Patternfly equivalents for these colors are available.

File: console/console-init/ui/src/stories/DeviceList.stories.tsx [hunk 8]
```diff
@@ -20,7 +20,7 @@ import { IRowData } from "@patternfly/react-table";
 import { DeviceDetailNavigation } from "modules/device-detail";
```
Comment (anukritijha) [inline console/console-init/ui/src/stories/DeviceList.stories.tsx:20]: License info missing.

File: console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+  const EnabledIcon = () => (
+    <Switch
+      id="device-enable-header-switch"
+      isChecked={isEnabled}
+      onChange={onChange}
+    />
+  );
+
+  const KebabOptionsLayout = () => {
+    const dropdownItems = [
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:127]: Type declaration missing.

File: console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+        <DeviceTitle />
+        <DeviceDetailInFlex />
+      </SplitItem>
+    </>
+  );
+  const onChange = () => {
+    changeEnable(!isEnabled);
+  };
+  const EnabledIcon = () => (
+    <Switch
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:119]: I believe there is a generic component available for Switch - SwitchWithToggle.

File: console/console-init/ui/src/stories/DeviceList.stories.tsx (hunk 6 above)
Comment (drcoolsanjeev) [reply console/console-init/ui/src/stories/DeviceList.stories.tsx:23]: Stories doesn't allow to export same title from two files.

File: console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+const DeviceDetailHeader: React.FunctionComponent<IDeviceDetailHeaderProps> = ({
+  deviceName,
+  addedDate,
+  lastTimeSeen,
+  onEdit,
+  onDelete,
+  onChange,
+  onClone
+}) => {
+  const DeviceTitle = () => (
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:62]: I don't think so we need separate component/function for small part of code. can you try to use this code   in return function straight forward.

File: console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+    <Split gutter="md">
+      <SplitItem>
+        <Title headingLevel="h1" size="4xl" id="device-name">
+          {deviceName}
+        </Title>
+      </SplitItem>
+    </Split>
+  );
+
+  const DeviceDetailInFlex = () => (
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:72]: I don't think so we need Flex postfix in function name. can we try something like DeviceDateSection

File: console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+      >
+        Last time seen :{" "}
+        <b>
+          <FormatDistance date={lastTimeSeen} />
+        </b>
+      </FlexItem>
+    </Flex>
+  );
+
+  const DeviceDetailLayout = () => (
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:92]: don't need separate component/function until and unless we don't have the good reason.

File: console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx (hunk 11 above)
Comment (anukritijha) [reply console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:62]: From my POV, the segregated code looks cleaner. :)

File: console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx (hunk 12 above)
Comment (anukritijha) [inline console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:72]: Can we revisit the name of DeviceDetailInFlex please?

Comment (anukritijha) [reply console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:72]: Like my earlier comment, as per my POV, the segregated code looks cleaner. :)

File: console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx (hunk 13 above)
Comment (anukritijha) [reply console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:92]: As stated earlier, from my POV, the segregated code looks cleaner. :)

File: console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+        <Title headingLevel="h1" size="4xl" id="device-name">
+          {deviceName}
+        </Title>
+      </SplitItem>
+    </Split>
+  );
+
+  const DeviceDetailInFlex = () => (
+    <Flex className={css(project_header_styles.namespace_info_margin)}>
+      <FlexItem id="device-added-date">
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:74]: Can we have the id in the decided format, as discussed?

**Comment by drcoolsanjeev** [comment]
@ajaypratap003 @anukritijha addressed all the comments

**Pull request merged**